File: paiguangguang/frontend/features/chatbot/types/message.ts
export type MessageRole = "user" | "assistant" | "system" | "tool";

export type MessageStatus = "pending" | "streaming" | "completed" | "failed" | "cancelled";

export type MessageData = {
  id: string;
  conversation_id: string;
  role: MessageRole;
  content: string;
  content_json: Record<string, unknown> | null;
  sequence_number: number;
  status: MessageStatus;
  model: string | null;
  parent_message_id: string | null;
  client_request_id: string | null;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  error_code: string | null;
  created_at: string;
  updated_at: string;
};

export type MessagePageData = {
  items: MessageData[];
  next_cursor: string | null;
  has_more: boolean;
};

export type ChatRequest = {
  content: string;
  client_request_id: string;
};

export type GenerationRequest = {
  client_request_id: string;
};

export type StopGenerationRequest = {
  assistant_message_id?: string | null;
};

export type StopGenerationData = {
  conversation_id: string;
  assistant_message_id: string;
  status: "cancellation_requested" | "cancelled" | "already_terminal";
};

File: paiguangguang/backend/app/chatbot/schemas/message.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MessageData(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    conversation_id: UUID
    role: str
    content: str
    content_json: dict[str, Any] | None = None
    sequence_number: int
    status: str
    model: str | None = None
    parent_message_id: UUID | None = None
    client_request_id: UUID | None = None
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    error_code: str | None = None
    created_at: datetime
    updated_at: datetime


class MessagePageData(BaseModel):
    items: list[MessageData]
    next_cursor: str | None = None
    has_more: bool

